### PR TITLE
docs: Add warning banners for dev versions on docs

### PR DESCRIPTION
**Pull request by kratsg**
# Pull Request Description

See #1286 and #1287. This adds banners at the top of the docs if the docs point to development docs or old versions.

For development docs, this checks to see if we're on `pyhf.readthedocs.io` or not. If we are not, then we're looking at latest (unstable) docs [e.g. on scikit-hep].

For old docs, if we are on `pyhf.readthedocs.io`, this will then make an asynchronous request to determine latest version of the docs and see if that matches what we're currently on.

Rendered pyhf.readthedocs.io docs here: https://pyhf.readthedocs.io/en/docs-versionwarnings/ (expect it to say "not a stable version" because it's on RTD and it's not the latest that RTD redirects to). This will also look different when pushed to scikit-hep as it will mention that this is the development version instead.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR
```
* Add banner to top to warn when docs being shown at scikit-hep.org/pyhf are development
   - Add custom CSS and JavaScript for Sphinx to use in docs/conf.py
```

### CODE REVIEW

**Comment by kratsg** [comment]
Screen Shot 2021-02-03 at 8 46 39 PM

File: docs/index.rst [hunk 1]
```diff
@@ -25,6 +25,13 @@
 
    <a class="github-fork-ribbon right-top fixed" href="https://github.com/scikit-hep/pyhf/" data-ribbon="View me on GitHub" title="View me on GitHub">View me on GitHub</a>
 
+
+.. raw:: html
+
+   <p id="dev-version"><strong>Warning:</strong> This is a development version. The latest stable version is at <a href="https://pyhf.readthedocs.io/">ReadTheDocs</a>.</p>
+
+   <p id="old-version"><strong>Warning:</strong> This is not a stable version. The latest stable version is <a id="latest-version-link" href="https://pyhf.readthedocs.io/">v?.?.?</a>.</p>
```
Comment (matthewfeickert) [inline docs/index.rst:33]:
```suggestion
   <p id="old-version"><strong>Warning:</strong> This is not the latest stable version. The latest stable version is <a id="latest-version-link" href="https://pyhf.readthedocs.io/">v?.?.?</a>.</p>
```

I think I'd still call old releases stable, but we just want to encourage using newer releases. So maybe just mention it is not the latest stable?

**Review by matthewfeickert — COMMENTED** [review]
@kratsg this is _really_ nice! Thank you! I just have one small change.

Also, did you write this all from scratch or did you find templating on another project? If there was another project that you were able to look at can you mention that in the PR body for reference?

**Comment by matthewfeickert** [comment]
@kratsg I turned on [Version warning in ReadTheDocs](https://docs.readthedocs.io/en/stable/versions.html#version-warning)

show_version_warning

which now show the following on older versions for RTD:

[warning_old_version](https://pyhf.readthedocs.io/en/v0.5.3/)

Do we still need the RTD logic in the `docs/_static/js/custom.js`? It seems no, and we could reduce this to just having to deal with `scikit-hep.org/pyhf`. Thoughts?

File: docs/_static/js/custom.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            url: "https://pyhf.readthedocs.io/",
+            success: function(data, textStatus, resp){
+                console.log(resp.getAllResponseHeaders());
+                const version = resp.getResponseHeader("x-rtd-version") || 'here';
+                if(window.location.href.indexOf(version) === -1){
+                    document.getElementById("latest-version-link").text = version;
+                    old_version.classList.add("version-warning");
+                }
+            }
+        });
```
Comment (matthewfeickert) [inline docs/_static/js/custom.js:27]: Do we still need this logic given https://github.com/scikit-hep/pyhf/pull/1288#issuecomment-773022449?

Comment (kratsg) [reply docs/_static/js/custom.js]: nope. removed.

**Comment by kratsg** [comment]
> Also, did you write this all from scratch or did you find templating on another project? If there was another project that you were able to look at can you mention that in the PR body for reference?

most of it is from scratch. I grabbed some styles based on the pallet theme, but it's pretty much all custom.

**Comment by matthewfeickert** [comment]
> most of it is from scratch. I grabbed some styles based on the pallet theme, but it's pretty much all custom.

:sunglasses:

**Review by matthewfeickert — APPROVED** [review]
@kratsg LGTM! This is an awesome PR. :smile:

**Pull request merged**